DQM/HLTEvF/plugins/HLTObjectMonitorHI.cc

```diff
@@ -0,0 +1,778 @@
+#include <memory>
+#include <sys/time.h>
+#include <cstdlib>
+
+// user include files
+#include "DataFormats/Common/interface/TriggerResults.h"
+#include "DataFormats/HLTReco/interface/TriggerObject.h"
+#include "DataFormats/HLTReco/interface/TriggerEvent.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "HLTrigger/HLTcore/interface/HLTConfigProvider.h"
+
+//for collections
+#include "HLTrigger/JetMET/interface/AlphaT.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/METReco/interface/MET.h"
+#include "DataFormats/RecoCandidate/interface/RecoChargedCandidate.h"
+#include "DataFormats/BTauReco/interface/JetTag.h"
+#include "DataFormats/RecoCandidate/interface/RecoChargedCandidateFwd.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/JetReco/interface/CaloJet.h"
+
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+
+#include "TMath.h"
+#include "TStyle.h"
+#include "TLorentzVector.h"
+
+#include <unordered_map>
+//
+// class declaration
+//
+
+//using namespace edm;
+using namespace trigger;
+using std::vector;
+using std::string;
+using std::unordered_map;
+
+class HLTObjectMonitorHI : public DQMEDAnalyzer {
+  struct hltPlot
+  {
+    
+    MonitorElement * ME;
+    string pathName;
+    string pathNameOR;
+    string moduleName;
+    string moduleNameOR;
+    int pathIndex = -99;
+    int pathIndexOR = -99;
+    string plotLabel;
+    string xAxisLabel;
+    int nBins;
+    double xMin;
+    double xMax;
+    bool displayInPrimary;
+    
+  };
+
+   public:
+      explicit HLTObjectMonitorHI(const edm::ParameterSet&);
+      ~HLTObjectMonitorHI() override;
+
+  //      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+
+   private:
+      void analyze(const edm::Event&, const edm::EventSetup&) override;
+      void bookHistograms(DQMStore::IBooker &i, edm::Run const&, edm::EventSetup const&) override;
+      void dqmBeginRun(edm::Run const&, edm::EventSetup const&) override;
+      void endRun(edm::Run const&, edm::EventSetup const&) override;
+      vector<hltPlot*> plotList;
+      //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+      //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+  double get_wall_time(void);
+      // ----------member data ---------------------------
+
+  bool debugPrint;
```

Comment:
please follow the coding rules: all member variables should be called either `m_variable` or `variable_`